File: Tyche.Android/app/src/main/java/com/pipel/tyche/pool/domain/Pool.kt
package com.pipel.tyche.pool.domain

import com.pipel.core.type.NonEmptyString
import com.pipel.core.type.PositiveInt
import com.pipel.core.type.Ulid

data class Pool(
    val poolId: Ulid,
    val poolLayoutId: Ulid,
    val poolName: NonEmptyString,
    val currentPosition: PositiveInt?,
    val beforePosition: PositiveInt?
)

File: Tyche.Backend/Pipel.Tyche.Pool/Pipel.Tyche.Pool/Domain/Pool.fs
namespace Pipel.Tyche.Pool.Domain

open Pipel.Tyche.Pool.Domain
open Pipel.Type

type PoolPK = { PoolId: Ulid }

type Pool =
    { PoolPK: PoolPK
      PoolLayoutPK: PoolLayoutPK
      PoolName: NonEmptyString100
      CurrentPosition: PositiveInt option
      BeforePosition: PositiveInt option }
